orbit/pkg/platform/platform_windows.go

```diff
@@ -341,3 +344,203 @@ func GetSMBiosUUID() (string, UUIDSource, error) {
 	// UUID was obtained from calling WMI infrastructure
 	return uuid, UUIDSourceWMI, nil
 }
+
+// getWorkingPath returns the current working directory
+func getWorkingPath() (string, error) {
+	// getting current executable fullpath
+	exec, err := os.Executable()
+	if err != nil {
+		return "", err
+	}
+
+	// returns the current executable directory
+	return filepath.Dir(exec), nil
+}
+
+// getOrbitVersion returns the version of the Orbit executable
+func getOrbitVersion(path string) (string, error) {
+	const (
+		expectedPrefix      = "orbit "
+		expectedVersionFlag = "-version"
+	)
+
+	if len(path) == 0 {
+		return "", errors.New("input executable is empty")
+	}
+
+	// running the executable with the version flag
+	args := []string{expectedVersionFlag}
+	out, err := exec.Command(path, args...).Output()
+	if err != nil {
+		return "", errors.New("there was a problem running target executable")
+	}
+
+	// parsing the output
+	versionOutputStr := string(out)
+	if len(versionOutputStr) == 0 {
+		return "", errors.New("empty executable output")
+	}
+
+	outputByLines := strings.Split(strings.TrimRight(versionOutputStr, "\n"), "\n")
+	if len(outputByLines) < 1 {
+		return "", errors.New("expected number of lines is not present")
+	}
+
+	rawVersionStr := strings.TrimSpace(strings.ToLower(outputByLines[0]))
+	if !strings.HasPrefix(rawVersionStr, expectedPrefix) {
+		return "", errors.New("expected version prefix is not present")
+	}
+
+	// getting the actual version string
+	versionStr := strings.ReplaceAll(rawVersionStr, expectedPrefix, "")
+	if len(versionStr) == 0 {
+		return "", errors.New("expected version information is not present")
+	}
+
+	return versionStr, nil
+}
+
+// fixSymlinkNotPresent fixes the issue where the symlink to the orbit service binary is not present
+// this is a workaround for the issue described here https://github.com/fleetdm/fleet/issues/10300
+func fixSymlinkNotPresent() error {
+	// getting current working directory
+	execPath, err := getWorkingPath()
+	if err != nil {
+		return err
+	}
+
+	// getting the path to orbit service binary
+	orbitPath := execPath + "\\..\\bin\\orbit\\orbit.exe"
+
+	// gathering target orbit version
+	versionOrbit, err := getOrbitVersion(orbitPath)
+	if err != nil {
+		return fmt.Errorf("getting orbit version: %w", err)
+	}
+
+	// checking if target orbit has the problematic logic
+	if versionOrbit != "1.6.0" && versionOrbit != "1.7.0" {
+		return nil
+	}
+
+	// checking if the orbit service binary symlink needs to be regenerated
+	_, err = os.Readlink(orbitPath)
+
+	// if there are no errors, it means that the symlink is present, nothing to do
+	if err == nil {
+		return nil
+	}
```

Comment:
Lines 407-432 are checking if the fix is needed.
They should be a separate function that can also be unit tested.

Reply:
thing is that quirks should be self-contained as much as possible so they can be treated as a single unit. In this case, 407-432 looks for specific conditions to this error

Reply:
Yes, but no reason not to make it a separate function 🤷 
Long functions are bad.